detect-pr-changes.sh

```diff
@@ -11,6 +11,7 @@ mkdir tmpgit/new
 
 # First build everything, so we can get straight to the good stuff at the end of the log.
 dotnet build tools/Google.Cloud.Tools.VersionCompat
```

Comment:
Do you still need this here? Wouldn't this one get built anyway because of the dependency CheckVersionCompatibility has on it?

Reply:
Yes, but I think I'd rather be explicit. If we ever remove the dependency in CheckVersionCompatibility (e.g. by making the two console apps depend on a single common library) it would be easy to forget to put this back.